packages/fxa-auth-server/lib/routes/utils/signin.js

```diff
@@ -398,33 +401,44 @@ module.exports = (log, config, customs, db, mailer) => {
           });
       }
 
-      function sendVerifyLoginCodeEmail() {
+      async function sendVerifyLoginCodeEmail() {
         log.info('account.token.code.start', {
           uid: accountRecord.uid,
         });
 
-        const geoData = request.app.geo;
-        return mailer
-          .sendVerifyLoginCodeEmail(accountRecord.emails, accountRecord, {
+        const secret = accountRecord.primaryEmail.emailCode;
+        const authenticator = new otplib.authenticator.Authenticator();
+        authenticator.options = Object.assign(
+          otplib.authenticator.options,
+          otpOptions,
+          { secret }
+        );
```

Comment:
This "create an `Authenticator` with the right options and generate a code" chunk repeats several times in our codebase now, I wonder if it's worth factoring out into a small helper module to reduce the risk of accidentally doing it differently in different locations.

Reply:
Cleaned this up a bit!